common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -116,12 +117,62 @@ public static async Task<ErrorCode> GetExceptionCodeAsync(HttpResponseMessage re
             try
             {
                 IoTHubExceptionResult responseContent = JsonConvert.DeserializeObject<IoTHubExceptionResult>(responseContentStr);
-                Dictionary<string, string> messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
 
-                if (messageFields != null
-                    && messageFields.TryGetValue(CommonConstants.ErrorCode, out string errorCodeObj))
+                try
                 {
-                    errorCodeValue = Convert.ToInt32(errorCodeObj, CultureInfo.InvariantCulture);
+                    Dictionary<string, string> messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
+
+                    if (messageFields != null
+                        && messageFields.TryGetValue(CommonConstants.ErrorCode, out string errorCodeObj))
+                    {
+                        errorCodeValue = Convert.ToInt32(errorCodeObj, CultureInfo.InvariantCulture);
```

Comment:
Curious if you considered `int.TryParse()` and if so, why you went with `Convert.ToInt32()`.

Reply:
I didn't really inspect this code, this was already in place. I'm seeing that int.TryPrase is supposed to be faster, and will also not throw an exception; so that seems like a better choice here.